### PR TITLE
capmt: add support for OSCam network protocol v1

### PR DESCRIPTION
**Pull request by manio**
As we discussed on IRC, I created this as a separate mode (which will be flexible in the future because of protocol versioning in the OSCam)
Currently the OSCam version is shown in the tvheadend log, and the tvheadend version in the OSCam log and web interface (when connected).

### CODE REVIEW

File: src/descrambler/capmt.c [hunk 1]
```diff
@@ -821,7 +850,17 @@ capmt_set_filter(capmt_t *capmt, int adapter, sbuf_t *sb, int offset)
...
   filter->pid = pid;
-  memcpy(&filter->filter, sbuf_peek(sb, offset + 8), sizeof(filter->filter));
+  if (capmt->capmt_oscam == CAPMT_OSCAM_NET_PROTO)
+  {
+    offset = 9; //filter data starts here
+    memcpy(filter->filter.filter, sbuf_peek(sb, offset), DMX_FILTER_SIZE);
+    offset += DMX_FILTER_SIZE;
+    memcpy(filter->filter.mask, sbuf_peek(sb, offset), DMX_FILTER_SIZE);
+    offset += DMX_FILTER_SIZE;
+    memcpy(filter->filter.mode, sbuf_peek(sb, offset), DMX_FILTER_SIZE);
```
Comment (perexg) [inline src/descrambler/capmt.c]: I think that one memcpy should be sufficient with 3 * DMX_FILTER_SIZE here.

File: src/descrambler/capmt.c [hunk 2]
```diff
@@ -777,14 +787,33 @@ capmt_service_destroy(th_descrambler_t *td)
 }
 
 static void
+capmt_send_client_info(capmt_t *capmt)
+{
+  char info_version[255];
+  snprintf(info_version, sizeof(info_version), "Tvheadend %s", tvheadend_version);
```
Comment (perexg) [inline src/descrambler/capmt.c]: Just cometic - one char[] should be enough. Just use buf + 7 and sizeof(buf) - 7 in this snprintf and put everything to this buffer:

*(uint32_t *)(buf + 0) = htonl(DVBAPI_CLIENT_INFO);
*(uint16_t *)(buf + 4) = htons(DVBAPI_PROTOCOL_VERSION);

The accesses seem aligned here..

File: src/descrambler/capmt.c [hunk 3]
```diff
@@ -854,7 +893,11 @@ capmt_stop_filter(capmt_t *capmt, int adapter, sbuf_t *sb, int offset)
 {
   uint8_t demux_index  = sbuf_peek_u8   (sb, offset + 4);
   uint8_t filter_index = sbuf_peek_u8   (sb, offset + 5);
-  int16_t pid          = sbuf_peek_s16be(sb, offset + 6);
+  int16_t pid;
+  if (capmt->capmt_oscam == CAPMT_OSCAM_NET_PROTO)
+    pid          = sbuf_peek_s16  (sb, offset + 6);
+  else
+    pid          = sbuf_peek_s16be(sb, offset + 6);
```
Comment (perexg) [inline src/descrambler/capmt.c]: Move code outside declaration block.

**Comment by manio** [comment]
Thank you for reviewing. Updated.

**Comment by perexg** [comment]
Merged. Thanks.

**Comment by nmaclean** [comment]
@manio - thanks, this seems to work for me. What is the benefit of this type of connection rather than the older ones?

**Comment by perexg** [comment]
@nmaclean : None, except that the TCP protocol stabilized a bit (solving endianess issues and server/client greeting messages to detect the future enhancements).

**Pull request closed without merging**